### PR TITLE
Deprecate get_env.h and move content to env.{h,c}

### PR DESCRIPTION
**Pull request by christophebedard**
Resolves these `TODO`s:

https://github.com/ros2/rcutils/blob/ca42bc9f588e11ad40f5879dcb9a364b435d2cba/include/rcutils/env.h#L30

https://github.com/ros2/rcutils/blob/ca42bc9f588e11ad40f5879dcb9a364b435d2cba/src/env.c#L26

I think the intention was originally to deprecate in Galactic and remove in H-turtle (https://github.com/ros2/rcutils/pull/250#issuecomment-628785186), but it missed the :boat: so it'll have to be deprecated in H-turtle and removed in I-turtle. I can create an issue for the removal after this is merged.

### CODE REVIEW

File: include/rcutils/env.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2020 Open Source Robotics Foundation, Inc.
+// Copyright 2017-2021 Open Source Robotics Foundation, Inc.
```
Comment (clalancette) [inline include/rcutils/env.h:1]: Let's not change this; we haven't been updating it anywhere else.  Same goes for all of the copyright dates below.

File: include/rcutils/env.h [hunk 2]
```diff
@@ -58,6 +55,67 @@ RCUTILS_WARN_UNUSED
...
+ * The c-string which is returned is only valid until the next time this
+ * function is called, because it is a direct pointer to the static storage.
+ * Also note that the string returned here should *not* be freed.
+ *
+ * The function first tries to get the HOME environment variable.
+ * If that variable exists and is non-empty, that will be returned.
+ * Otherwise, the function tries to get the USERPROFILE environment variable.
+ * If that variable exists and is non-empty, that will be returned.
+ * If neither exists, NULL will be returned.
+ * The above algorithm is portable across both Unix and Windows.
```
Comment (clalancette) [inline include/rcutils/env.h:106]: I know that you are just moving this, but this documentation is actually incorrect.  On Unix, `rcutils_get_home_dir` will try to get the HOME environment variable only.  If that fails, it returns NULL.  On Windows, `rcutils_get_home_dir` will try to get the HOME environment variable.  If that fails, it tries to get the USERPROFILE environment variable.  If that fails, it returns NULL.

File: include/rcutils/get_env.h [hunk 3]
```diff
@@ -17,77 +17,14 @@
...
-const char *
-rcutils_get_home_dir(void);
-
-#ifdef __cplusplus
-}
+#ifdef _MSC_VER
+#pragma message ("rcutils/get_env.h has been deprecated, please include rcutils/env.h instead")
+#else
+#warning rcutils/get_env.h has been deprecated, please include rcutils/env.h instead
 #endif
```
Comment (clalancette) [inline include/rcutils/get_env.h:26]: Do we really need a separate `#pragma` on Windows?  Looking around the codebase, there are a couple of other instances that just unconditionally use `#warning`, which leads me to believe it is supported by MSVC.  But I'm not sure.

**Review by clalancette — CHANGES_REQUESTED** [review]
A few minor things to fix up, but this is a good cleanup.  Thanks!

File: include/rcutils/get_env.h (hunk 3 above)
Comment (christophebedard) [reply include/rcutils/get_env.h:26]: Yeah I was wondering too. `#warning` is a `gcc` extension, and apparently to get an actual/proper warning on MSVC we have to use `#pragma message`.

In any case, I can just use `#warning` if you prefer.

File: include/rcutils/env.h (hunk 1 above)
Comment (christophebedard) [reply include/rcutils/env.h]: Alright, I reverted 4eb6f857ba63e3e26b8f2384c10031cff7f3a399: fed8ca8fdf7094adb99ce2e936162c64f78dfffa

File: include/rcutils/env.h (hunk 2 above)
Comment (christophebedard) [reply include/rcutils/env.h]: Fixed in 7eb69e04cecd44bd3bf0a47c38bae5b05a873ba3

**Review by fujitatomoya — APPROVED** [review]
looks good to me

File: include/rcutils/get_env.h (hunk 3 above)
Comment (clalancette) [reply include/rcutils/get_env.h:26]: > Yeah I was wondering too. `#warning` is a `gcc` extension, and apparently to get an actual/proper warning on MSVC we have to use `#pragma message`.
> 
> In any case, I can just use `#warning` if you prefer.

Ah, I didn't realize.  OK, in that case we can leave this in place, I was just wondering why it was different.

**Review by clalancette — APPROVED** [review]
Looks good to me with green CI.

**Comment by christophebedard** [comment]
Should I do a CI run for this PR + the other related PRs all at once?

**Comment by clalancette** [comment]
> Should I do a CI run for this PR + the other related PRs all at once?

Yeah, that would be my suggestion.  And in this case, since we are touching so many different packages, I'll just suggest doing one run where we build and test everything.

**Comment by christophebedard** [comment]
Test failures on macOS look unrelated.

**Comment by clalancette** [comment]
> Test failures on macOS look unrelated.

Yeah, agreed.  Annoying, because it looks new, but not related to this PR.

**Pull request merged**